Assets/GameAssets/Player/PlayerScripts/PlayerHealth.cs

```diff
@@ -53,9 +53,23 @@ public override void TakeDamage(float damage, IBaseDamage damageSource)
 
             RuntimeSfxManager.APlayOneShotSfx?
                 .Invoke(FmodSfxEvents.Instance.PlayerTookDamage, gameObject.transform.position);
+
+            _iFrameCoroutine = StartCoroutine(InvincibilityFrames());
         }
     }
 
+    /// <summary>
+    /// used to grant the player temporary invincibility after taking damage
+    /// </summary>
+    /// <returns></returns>
+    private IEnumerator InvincibilityFrames()
+    {
+        _shouldTakeDamage = false;
+        yield return new WaitForSeconds(_iFrameDelayInSeconds);
```

Comment:
Not a requirement but this can be optimized a bit, there is a thread in lv-engineering (scroll up a little bit) called Tween Optimization, I talked about how coroutines can be improved in there